core/shared/src/main/scala-2.13+/zio/ChunkFactory.scala

```diff
@@ -31,4 +30,9 @@ private[zio] trait ChunkFactory extends StrictOptimizedSeqFactory[Chunk] {
         iterableOnce.iterator.foreach(chunkBuilder.addOne)
         chunkBuilder.result()
     }
+
+  final protected def fromArraySeq[A](seq: mutable.ArraySeq[A]): Chunk[A] = {
+    val arr = seq.array
+    Chunk.fromArray(Array.copyAs(arr, arr.length)(seq.elemTag)).asInstanceOf[Chunk[A]]
```

Comment:
I'm very surprised the `fromArray` method doesn't copy/require `Unsafe`...

Reply:
Yeah the `Unsafe` implicit would actually be good to have. However that wouldn't be binary compatible :(